src/debug/ee/arm64/arm64walker.cpp

```diff
@@ -2,21 +2,460 @@
+    //}
+
+    return patchBypass;
+}
+
+//Decodes PC Relative Branch Instructions
+//This code  is shared between the NativeWalker and DebuggerPatchSkip.
+//So ENSURE THIS FUNCTION DOES NOT CHANGE ANY STATE OF THE DEBUGEE
+//This Function Decodes :
+// BL     offset
+// B      offset
+// B.Cond offset
+// CB[N]Z X<r> offset
+// TB[N]Z X<r> offset
+
+//Output of the Function are:
+//offset - Offset from current PC to which control will go next
+//WALK_TYPE
+
+BOOL  NativeWalker::DecodePCRelativeBranchInst(PT_CONTEXT context, const PRD_TYPE& opcode, PCODE& offset, WALK_TYPE& walk)
+{
+#ifdef _DEBUG
+    PCODE incomingoffset = offset;
+    WALK_TYPE incomingwalk = walk;
+#endif
+
+    if ((opcode & 0x7C000000) == 0x14000000) // Decode B & BL
+    {
+        offset = (opcode & 0x03FFFFFF) << 2;
+        // Sign extension
+        if ((offset & 0x4000000)) //Check for 26'st bit
+        {
+            offset = offset | 0xFFFFFFFFF8000000;
+        }
+
+        if ((opcode & 0x80000000) != 0) //BL
+        {
+            walk = WALK_CALL;
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to BL %p \n", opcode, offset));
+        }
+        else
+        {
+            walk = WALK_BRANCH; //B
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to B %p \n", opcode, offset));
+        }
+        return TRUE;
+    }
+
+    //Conditional Branches
+    _ASSERTE(context != NULL);
+   
+
+    if ((opcode & 0xFF000010) == 0x54000000) // B.cond
+    {
+        WORD cond = opcode & 0xF;
+        bool result = false;
+        switch (cond >> 1)
+        {
+        case 0x0:  result = (context->Cpsr & NZCV_Z) != 0; // EQ or NE
+            break;
+        case 0x1:  result = (context->Cpsr & NZCV_C) != 0; // CS or CC
+            break;
+        case 0x2:  result = (context->Cpsr & NZCV_N) != 0; // MI or PL
+            break;
+        case 0x3:  result = (context->Cpsr & NZCV_V) != 0; // VS or VC
+            break;
+        case 0x4:  result = ((context->Cpsr & NZCV_C) != 0) && ((context->Cpsr & NZCV_Z) == 0); // HI or LS
+            break;
+        case 0x5:  result = ((context->Cpsr & NZCV_N) >> NZCV_N_BIT) == ((context->Cpsr & NZCV_V) >> NZCV_V_BIT); // GE or LT
+            break;
+        case 0x6:  result = ((context->Cpsr & NZCV_N) >> NZCV_N_BIT) == ((context->Cpsr & NZCV_V) >> NZCV_V_BIT) && ((context->Cpsr & NZCV_Z) == 0); // GT or LE
+            break;
+        case 0x7:  result = true; // AL
+            break;
+        }
+
+        if ((cond & 0x1) && (cond & 0xF) != 0) { result = !result; }
+
+        if (result)
+        {
+            walk = WALK_BRANCH;
+            offset = ((opcode >> 5) & 0x7FFFF)<<2; //imm19:00 -> 21 bits
+
+            if (offset & 0x100000) //sign extension check for 21's bit
+            {
+                offset = offset | 0xFFFFFFFFFFE00000;
+            }
+            LOG((LF_CORDB, LL_INFO100000, "Arm64Walker::Decoded opcode: %x to B.cond %p \n", opcode, offset));
```

Comment:
The above would be useful if it were extracted to a function as LDR literal needs to do this as well